### DISCUSSION TITLE
Support for Shelly Mini?

### DISCUSSION THREAD

**Discussion by perstromqvist** [mongoose-os-apps/shelly-homekit#1499] [Ideas]
Will the new Shelly Mini's be (listed as) supported by Mongoose?

**Comment by rionshin** [comment]
@timoschilling  Does the new mini devices use same chip, do you plan to support it ? 

https://www.shelly.com/en-bg/products/shop/shelly-plus-1-mini

**Comment by timoschilling** [comment]
I don’t have a test device, but possibly it will. The mini 1 would be a problem I think. The mini EM and mini 1pm would be harder.

**Comment by jkdeleteme** [comment]
+1

**Comment by Larsn1** [comment]
+1

**Comment by petereit** [comment]
In the mean time, I added my 1 Plus Mini's to HomeKit by installing Homebridge and Mosquitto MQTT broker on an old Mac Mini that I use as an AirPrint server.

**Comment by Larsn1** [comment]
Well I think that’s a good solution bc the plus minis get matter if Shelly releases a firmware update for it

**Comment by cpm1** [comment]
> I don’t have a test device, but possibly it will. The mini 1 would be a problem I think. The mini EM and mini 1pm would be harder.

I'd be more than happy to sponsor both a Shelly Plus 1 Mini as well as a Shelly Plus 1PM Mini to you :)

**Comment by Larsn1** [comment]
Updated mine to Tasmota 13.2 which supports matter. Is the method of injecting the bootloader really so different?

**Comment by juliennnnn** [comment]
Hello,

Do you plan to support the Shelly Plus 1 Mini ?

Thanks a lot

**Comment by lucassite120** [comment]
Finally, I have my Shelly Mini working in Homekit with HAA firm: https://github.com/RavenSystem/esp-homekit-devices

I followed these steps: https://www.youtube.com/watch?v=06YHkRkwJE4

**Comment by MicVog1412** [comment]
+1

**Comment by timoschilling** [comment]
> I'd be more than happy to sponsor both a Shelly Plus 1 Mini as well as a Shelly Plus 1PM Mini to you :)

@cpm1 write me a email, address can be found on my profile.

**Comment by cpm1** [comment]
> write me a email, address can be found on my profile.

@timoschilling FYI, did so on 21st of December. If you didn't get it, please check your spam drawer.

**Comment by Schw3n** [comment]
> > write me a email, address can be found on my profile.
> 
> @timoschilling FYI, did so on 21st of December. If you didn't get it, please check your spam drawer.

I also sent a mail on 21st Dec. to offer also a sponsoring.

**Comment by skyedrummer** [comment]
Any news on the Mini support? Currently using HB for them, but this is best.

**Comment by SchroedersKater** [comment]
If it helps, I’d like to sponsor a Shelly 1PM Mini Gen3 as well. Just drop me a PM. BTW, where are you located? To some countries shipping costs are more than the device costs.

**Comment by DHEVAN1** [comment]
Hi, Just wanted to know if there is any update on support for Minis, and if you think its possible to happen.

**Comment by g3ntleman** [comment]
I accidentally ordered a "1 Mini, gen3". Specs say it has a "ESP-Shelly-C38F" CPU and 8 MB of memory as opposed to the 1 Plus Mini (4 MB).

From [support forum](https://community.shelly.cloud/topic/661-shelly-c3f8-specification):

All Gen3 devices are based on Shelly-C3F8 MCU. The Shelly-C38F are fully compatible with ESP32-C3, but with build-in more Flash and RAM which allow to save space into devices and  give us possibility to add more features.

**Comment by lukasfoempe** [comment]
Hi all,
is there any update for supporting the 1mini gen3?
Would be great.

br
lukas

**Comment by I3lackJ4ck** [comment]
@timoschilling is there any news on adding the new Shelly 1 Mini Gen 3 to the list of Homekit support?

**Comment by DHEVAN1** [comment]
> @timoschilling is there any news on adding the new Shelly 1 Mini Gen 3 to the list of Homekit support?

I have an idea and I haven't bought a mini gen 3 yet but if you would like you can try it. You can. You can add it to home assistant and have home assistant running. Then you can get the HomeKit plugin on home assistant which should make your home assistant devices work on HomeKit.

**Comment by tobz-nz** [comment]
> I have an idea and I haven't bought a mini gen 3 yet but if you would like you can try it. You can. You can add it to home assistant and have home assistant running. Then you can get the HomeKit plugin on home assistant which should make your home assistant devices work on HomeKit.

I've done this, and its working great. I've got 3 mini gen 3's and a H&T sensor all in HomeKit via HA

**Comment by I3lackJ4ck** [comment]
Hello,
thanks for the answer. 
I want to avoid having to setup a Homeassistant or homebridge with tasmota or anything like that as i have no need for more usage other than a switch. I already have 10 shellys with the mongoose implementation and they all work great. Seems like i have to wait or try to get one of the older Shelly 1s

**Comment by DHEVAN1** [comment]
> > I have an idea and I haven't bought a mini gen 3 yet but if you would like you can try it. You can. You can add it to home assistant and have home assistant running. Then you can get the HomeKit plugin on home assistant which should make your home assistant devices work on HomeKit.
> 
> 
> 
> I've done this, and its working great. I've got 3 mini gen 3's and a H&T sensor all in HomeKit via HA

Can you send a picture of what the HomeKit interface looks like with the Shelly's on there. I am planning on buying 10+ Shelly mini gen 3's @tobz-nz

**Comment by tobz-nz** [comment]
@DHEVAN1 This is one room. The `Lights` is a Shelly mini & the Temp & Humidity are the H&T sensors.
The heater and lamp are other things.

<img width="354" alt="Screenshot 2024-06-06 at 2 00 01 PM" src="https://github.com/mongoose-os-apps/shelly-homekit/assets/443054/2b680bf7-ba79-4be3-abb2-dbb301eabf46">

**Comment by metaljay** [comment]
Is this even being worked on anymore? I only ask as I can’t see a development branch or any comments regarding support for the Gen3 products.

**Comment by markirb** [comment]
Hi,

I do not think someone (@timoschilling ?) actively pursues Mini Support at this time. Though it should most likely not be that hard to add support, this takes time and resources which are limited for us currently .)

If you want to support, then you can do so with (in order):  Writing Code, Donating Hardware, Testing, Money.... 

We will release a new version soon (Support for +2PM), and maybe Shelly Gen3 is the next step once this is completed.

**Comment by metaljay** [comment]
> Hi,
> 
> I do not think someone (@timoschilling ?) actively pursues Mini Support at this time. Though it should most likely not be that hard to add support, this takes time and resources which are limited for us currently .)
> 
> If you want to support, then you can do so with (in order): Writing Code, Donating Hardware, Testing, Money....
> 
> We will release a new version soon (Support for +2PM), and maybe Shelly Gen3 is the next step once this is completed.

I can help with two of those things - money and donating hardware.

I’ve already emailed @timoschilling offering money as I personally have used his firmware many times. 
If there is anyone else actively giving this ago (and willing to do a pull request) I’ll happily donate to you instead

**Comment by markirb** [comment]
Hi.

Let's talk after the current release, I hope this weekend.

Maybe Timon will give it a go, maybe I can do it. Let's see

**Comment by lucassite120** [comment]
I used HAA HomeKit firmware with my Shelly Minis. It's very easy to install using OTA and works great. Check it out:
https://github.com/RavenSystem/mgos32toHAA

**Comment by iiseppi** [comment]
When testing to support Shelly Mini's start, does that mean also that, new GEN3 devices is getting support?

**Comment by markirb** [comment]
It would mean that the way to support other devices is not far.

@metaljay I sent you an email.

**Comment by markirb** [comment]
Looking good.... first version is running. 

Things not working so far:
Temperature NTC (library not yet compatible with esp32c3)
PWM (library needs to be ported; not needed actually)

Still some work to do, especially to incorporate the workarounds into the normal build environment.
If anyone wants to try (use at your own risk). Revert did work for me to and from 1.3.3:


[fw.zip](https://github.com/user-attachments/files/16254817/fw.zip)

**Comment by metaljay** [comment]
> Looking good.... first version is running.
> 
> Things not working so far: Temperature NTC (library not yet compatible with esp32c3) PWM (library needs to be ported; not needed actually)
> 
> Still some work to do, especially to incorporate the workarounds into the normal build environment. If anyone wants to try (use at your own risk). Revert did work for me to and from 1.3.3:
> 
> [fw.zip](https://github.com/user-attachments/files/16254817/fw.zip)

Posts like these bring me so much happiness (sad I know)
You only got the minis the other day and you’ve already got a working version! Well done!!

**Comment by markirb** [comment]
I mean, it was really not hard to do. It just worked...

Which is surprising since it is RISC-V now. But rojers lasts commits for mongoose-os that has since been kind of abandoned, did most of the work...

**Comment by DHEVAN1** [comment]
Hey whenever this is being worked on will the Shelly app still work after the software is finalized? Like if I wanted to check power monitoring or would it not work?On Jul 16, 2024, at 4:08 PM, Markus Kirberg ***@***.***> wrote:﻿
I mean, it was really not hard to do. It just worked...
Which is surprising since it is RISC-V now. But rojers lasts commits for mongoose-os that has since been kind of abandoned, did most of the work...

**Comment by markirb** [comment]
This app never has and almost certainly never will support Shelly App/Cloud

**Comment by DHEVAN1** [comment]
Alright thanks!On Jul 17, 2024, at 9:12 AM, Markus Kirberg ***@***.***> wrote:﻿
This app never has and almost certainly never will support Shelly App/Cloud

**Comment by lucassite120** [comment]
Since [HAA firmware](https://github.com/RavenSystem/mgos32toHAA) adds HomeKit suport to all new Shellys with infinite configurations, having Shelly App/Cloud with Mongoose would be a great feature.

**Comment by metaljay** [comment]
> Since [HAA firmware](https://github.com/RavenSystem/mgos32toHAA) adds HomeKit suport to all new Shellys with infinite configurations, having Shelly App/Cloud with Mongoose would be a great feature.

Yeh that’s great, but not something they are looking at (and I don’t blame them for the niche population that want it). I’d much rather have this firmware than the HAA one. If you’ve used this firmware on Shelly 1s before vs the HAA one you’ll know how lightweight this is and how the response times are incredible.

**Comment by lucassite120** [comment]
> Yeh that’s great, but not something they are looking at (and I don’t blame them for the niche population that want it). I’d much rather have this firmware than the HAA one. If you’ve used this firmware on Shelly 1s before vs the HAA one you’ll know how lightweight this is and how the response times are incredible.

I have used both firmwares, and HAA is more light and has better response time by far.

**Comment by metaljay** [comment]
> I have used both firmwares, and HAA is more light and has better response time by far.

Then don’t clog up a discussion promoting another firmware (that ultimately requires a paid app on iPhone / android to configure easily).

**Comment by markirb** [comment]
unfortunately all that's @lucassite120 is doing: spamming several topics promoting HAA.... @lucassite120 I do not know what your interest is, but I start to doubt your only motive is to help other users so please stop it

**Comment by lucassite120** [comment]
I am not promoting HAA. I was just trying to help users because this firmware is practically dead since Rojer left. I have tried the new version and it is even slower than the latest Rojer's version. I have used this firmware since the beginning, but it is a pity what it is now. I only asked about Shelly's cloud feature because it is something that HAA will never have. I have over 120 Shelly devices and using the Shelly app along with HomeKit would be a great point. Don't worry, I'll leave this community of fanatics. The HAA community is much friendlier. FYI: HAA App is not needed, I only use it to batch updates. ASAP I will move to HAA the rest of my Shellys. Bye!

**Comment by markirb** [comment]
You may have the feeling that it is slower. Do you have numbers? I doubt it to be slower since nothing changed underneath. 

Mongoose OS is still the same. And even basically the same that Shelly itself uses until now based on IDF 4.4. Shelly seem to continue their very own mos fork now. But only in 1.4.0 (currently in beta) they migrate to newer IDF.

Users can choose whatever they want, there are several firmware alternatives and each have their up- and downsides. If you migrate for not liking this community then fine. 
I only feel that it is not very helpful if you post under every
Thread what you use. 

If you want this firmware to be faster and better then any help is appreciated.

**Comment by danielowenuk** [comment]
@markirb I loaded your firmware to a mini1, appears to be working perfectly

I did have a few linked to homebridge I always prefer native homekit where possible.

VERY much appreciated.

**Comment by skyedrummer** [comment]
Does this firmware work on Gen3 devices?

**Comment by metaljay** [comment]
> Does this firmware work on Gen3 devices?

There’s a beta firmware that you can compile yourself that supports it, or wait a few week/s and it will be published as general release (there’s an active pull request ongoing now)

**Comment by skyedrummer** [comment]
Try as I might I can't get the fw.zip to load into the shelly mini from its GUI. Mini is on 1.3.3. If I update shelly to 1.4.0 beta, it flashes that it is uploading to the mini and then nothing. Any ideas?

**Comment by markirb** [comment]
Update from 1.4.0 is currently not possible. I have a compatible brach, but that's nothing for now. Go back to 1.3
fw.zip from above is only compatible with mini1 gen3

**Comment by skyedrummer** [comment]
Ah, okay, I think the mini I’m trying to update is not a gen3 device. Will the finished firmware be compatible with all the versions?

**Comment by markirb** [comment]
Was not planned. But will be very easy. Cannot promise but maybe after my summer vacation

**Comment by markirb** [comment]
Support for Minis is there in latest beta available in releases section: Mini 1 Gen3, Mini 1PM Gen3 and also Plus 1 Mini.
Please test

**Comment by pfrasun** [comment]
Just tested 2.12.2_beta1 with Shelly Plus 4 - after update it misses INPUTS in UI.

**Comment by markirb** [comment]
> Just tested 2.12.2_beta1 with Shelly Plus 4 - after update it misses INPUTS in UI.

Thanks you're right, totally unrelated javascript error. Is fixed in beta2 which is up now

**Comment by skyedrummer** [comment]
For the life of me I can't get the firmware to upload to the shelly plus1 mini. Can you post a link to the new firmware and maybe some instructions to flash the plus1 mini? I'm not even sure I have the correct firmware downloaded.

**Comment by markirb** [comment]
Which file do you have? You have to update from 1.3.3. from 1.4 it is not possible atm

**Reply by wmenasco** [reply]
> Which file do you have? You have to update from 1.3.3. from 1.4 it is not possible atm

Have an eta when 1.4.2 will supported?

**Comment by skyedrummer** [comment]
I downloaded this file  - shelly-homekit-ShellyPlus1Mini.zip  
from
https://github.com/mongoose-os-apps/shelly-homekit/releases/tag/2.12.2-beta1.

I'm on 1.3.3 - should I upgrade to 1.4.2? or do you mean go to 1.3.3 from 1.4?

edit - ok, I got mixed up with your response, I can only update from 1.3.3 
have i got the correct firmware file though?

**Comment by markirb** [comment]
> edit - ok, I got mixed up with your response, I can only update from 1.3.3 have i got the correct firmware file though?

Yes this should be the correct firmware. You upload this zip in the firmware section of the shelly. If this does not work then please enable debug over WebSocket and Watch Logs in Diagnostics.

**Comment by skyedrummer** [comment]
here is the log output (if it works)

[diagnostics-shelly-plus1mini-debug-log.txt](https://github.com/user-attachments/files/16825871/diagnostics-shelly-plus1mini-debug-log.txt)

**Comment by markirb** [comment]
> here is the log output (if it works)
> 
> 
> 
> [diagnostics-shelly-plus1mini-debug-log.txt](https://github.com/user-attachments/files/16825871/diagnostics-shelly-plus1mini-debug-log.txt)
> 
> 

Thanks. That does help indeed. Will need to build a new version

**Comment by markirb** [comment]
This here should work: 
[shelly-homekit-ShellyPlus1Mini.zip](https://github.com/user-attachments/files/16836725/shelly-homekit-ShellyPlus1Mini.zip)

**Comment by skyedrummer** [comment]
Thats seems to have done the trick, brilliant, thanks for all your work!

**Comment by Monst3r22** [comment]
I used your Shelly1miniGen3 firmware. At first thanks for your work!
Flashing worked and the Shelly is connected with HomeKit. However it doesn't switch the actual device anymore. It's connected to a light. The switch in HomeKit shows on/off but the bulb itself stays off all the time.

**Reply by markirb** [reply]
Shelly 1 Mini Gen3 definitely works. Do you have the PM Version? Does it work when you downgrade to stock?

**Reply by Monst3r22** [reply]
No, regular Shelly 1 Mini Gen3. Just went back to stock firmware which is working fine. Then back to HomeKit and again no response to the toggle.

**Reply by markirb** [reply]
So you say the output is not working or the input?

**Reply by Monst3r22** [reply]
No external switch connected. The shelly is controlled by the app and is supposed to switch a light.
So I guess the output is not working.

**Reply by markirb** [reply]
It is definitely the right GPIO at least for the version I have here. What does your case say? Ver1?

I could only explain that Relais is stuck. Can you try to toggle really often?

**Reply by Monst3r22** [reply]
Tried a new Mini 1. This one worked, so it was a faulty device. Thanks again!

**Reply by markirb** [reply]
That's makes it really strange. How should this work on original firmware then on the other device

**Reply by Monst3r22** [reply]
Strange indeed. But I tried HomeKit vs. Shelly back and forth several times with the given result. The new one worked straight after flashing.

**Reply by markirb** [reply]
It could be the case that the relay needs a greater current strength to be able to switch on or off; that would be the only thing that could be adjusted. 

You say it is not turning off yes? Does it turn on at the start and is always on?

**Reply by GuillermoMontosa** [reply]
> It could be the case that the relay needs a greater current strength to be able to switch on or off; that would be the only thing that could be adjusted.
> 
> You say it is not turning off yes? Does it turn on at the start and is always on?

I have recently bought two Shelly 1 Mini Gen 3. The same thing happens to me. Everything works fine with the Shelly firmware, however, once I flash to Mongoose OS, the Shelly cannot be turned on in Home Kit. It stays off. It is connected to the network, but there is no response from the Shelly. I flashed back to Shelly OS and the switch turns on and off.

So, there is something definetly wrong with this new version, it was not a faulty Shelly.

**Reply by markirb** [reply]
This version is partly correct. However it needs some yet unknown tweak or setting to work for every mini gen3 out there. I‘m running out of ideas currently as it does not seem to be the driver strength…

**Reply by GuillermoMontosa** [reply]
> Strange indeed. But I tried HomeKit vs. Shelly back and forth several times with the given result. The new one worked straight after flashing.

Was it exactly the same Shelly unit? 1 Mini Gen 3? What firmware did you flash Mongoose into, 1.3.3 or 1.4.2?

**Reply by skyedrummer** [reply]
I'm finding that if there is a HomePod involved in the HomeKit setup, it is really cranky. I quite often have to restart the HomePod and if there is an Apple TV involved, wait a few minutes till HomeKit switches over to using it as the hub. Then the Gen3 Mini works with no issues. Just a thought...

**Reply by GuillermoMontosa** [reply]
I don't have a HomePod, just Apple TV.

**Comment by Monst3r22** [comment]
1 mini Gen3 from 1.3.3
When I updated to 1.4.2 the flashing to mongoose didn‘t work anymore and I changed to the new Shelly which worked out of the box with 1.3.3

**Reply by GuillermoMontosa** [reply]
The previous Shelly, the one that didn't work, did you happen to update its firmware to 1.4.2, then upon finding Homekit flashing didn't work reverted back to 1.3.3 and then tried Mongoose again?

And, the new Shelly, the one that worked straight out the box.. by any chance you didn't update to 1.4.2 and stayed on 1.3.3 from the get go?
